.github/workflows/pages_publish.yml

```diff
@@ -0,0 +1,35 @@
+name: Publish
+on:
+  release:
+    types: [created]
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
```

Comment:
what is this permission used for?

Reply:
It is needed so we can publish to github ages.